libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorAdaptiveRecognizer.cs

```diff
@@ -0,0 +1,280 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using AdaptiveExpressions.Properties;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.Dialogs.Adaptive.Recognizers;
+using Microsoft.Bot.Builder.TraceExtensions;
+using Microsoft.Bot.Schema;
+using Microsoft.Orchestrator;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.AI.Orchestrator
+{
+    public class OrchestratorAdaptiveRecognizer : Recognizer
+    {
+        [JsonProperty("$kind")]
+        public const string Kind = "Microsoft.OrchestratorRecognizer";
+
+        /// <summary>
+        /// Intent name that will be produced by this recognizer if the child recognizers do not have consensus for intents.
+        /// </summary>
+        private const string ChooseIntent = "ChooseIntent";
+
+        /// <summary>
+        /// Property name for candidate intents that meet the ambiguity threshold.
+        /// </summary>
+        private const string CandidatesCollection = "candidates";
+
+        private static string modelPath = null;
+        private string snapshotPath = null;
+        private OrchestratorRecognizer recognizer = null;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OrchestratorAdaptiveRecognizer"/> class.
+        /// </summary>
+        /// <param name="callerLine">caller line.</param>
+        /// <param name="callerPath">caller path.</param>
+        [JsonConstructor]
+        public OrchestratorAdaptiveRecognizer([CallerFilePath] string callerPath = "", [CallerLineNumber] int callerLine = 0)
+            : base(callerPath, callerLine)
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets the full path to the NLR model to use.
+        /// </summary>
+        /// <value>
+        /// Model path.
+        /// </value>
+        [JsonProperty("modelPath")]
+        public StringExpression ModelPath { get; set; }
```

Comment:
>ModelPath [](start = 32, length = 9)

Should this be ModelUrl?  So they can point to cloud blobs?

Reply:
A good E2E here would be for the bot developer to use `bf orchestrator` command group to get an NLR model, create snapshots and push them up with the bot. Do you have scenarios in mind that would necessitate these to be Urls? The runtime impact to downloading models and or snapshots on the fly would be significant. In the skills case, I would expect an OrchestratorVACore that can ingest snapshots on the fly where the snapshot was constructed from LU content (Url to LU content).

Reply:
my assumption is that we don't want to check binaries into the repo.  Somebody somewhere has to pull them onto the box.

Reply:
@tomlm yes. the workflow is this - to get their machine set up, developer
1. runs `bf orchestrator:nlr:get` to download a specific model version to a gitignored folder
2. runs `bf orchestrator:create` or `bf orchestrator:build` to generate `.blu` and `.dialog` files. Keep these gitignored as well
3. [if not using build command and declarative dialogs] updates settings (or .env) with path to the model and .blu file

Developer then can deploy the model + blu files to app service or use `bf orchestrator` command in the pipelines.

This workflow would mean nothing is downloaded on the fly.